subs/pantry/src/Pantry/Types.hs

```diff
@@ -1002,12 +1005,56 @@ parsePackageIdentifier str =
 parsePackageName :: String -> Maybe PackageName
 parsePackageName = Distribution.Text.simpleParse
 
+-- | A package name parse fail.
+data PackageNameParseFail
```

Comment:
I'd rather include these data constructors with the existing `PantryException` data type, otherwise the docs there will be wrong.